components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -4813,4 +4816,19 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
             userId: user.id,
         });
     }
+
+    async adminGetBlockedEmailDomains(ctx: TraceContextWithSpan): Promise<EmailDomainFilterEntry[]> {
+        const user = this.checkAndBlockUser("adminGetBlockedEmailDomains");
+        await this.guardAdminAccess("adminGetBlockedEmailDomains", { id: user.id }, Permission.ADMIN_USERS);
+        return await this.emailDomainFilterdb.getFilterEntries();
```

Comment:
without pagination? there are 7979 entries. this could lead to a perf problem, at least on frontends.